addons/web/static/src/legacy/js/core/translation.js

```diff
@@ -73,6 +73,9 @@ var TranslationDataBase = Class.extend(/** @lends instance.TranslationDataBase#
         }
         if (lang) {
             paramsGet.lang = lang;
+        } else if (session.is_frontend && session.lang_url_code) {
+            // Keep distinct cached responses per language.
+            paramsGet.unique = session.lang_url_code;
```

Comment:
@odoo/rd-framework-js Do you want to review this or should we proceed internally?
(The rest of the PR is just a test tour, this is the real codebase change of the PR)

Reply:
@rdeodoo it looks acceptable to me. I think you can proceed internally